weave-js/src/components/PagePanelComponents/Home/Browse3/filters/FilterBar.tsx

```diff
@@ -143,29 +152,36 @@ export const FilterBar = ({
     (field: string) => {
       const defaultOperator = getOperatorOptions(field)[0].value;
       const newModel = {
-        ...filterModel,
+        ...localFilterModel,
         items: [
-          ...filterModel.items,
+          ...localFilterModel.items,
           {
-            id: filterModel.items.length,
+            id: localFilterModel.items.length,
             field,
             operator: defaultOperator,
           },
         ],
       };
-      setFilterModel(newModel);
+      setLocalFilterModel(newModel);
     },
-    [filterModel, setFilterModel]
+    [localFilterModel]
+  );
+
+  const debouncedSetFilterModel = useMemo(
+    () =>
+      _.debounce((newModel: GridFilterModel) => setFilterModel(newModel), 300),
```

Comment:
make constant?

Reply:
yeah, and I would increase the rate to more like 1 sec

Reply:
that adds 1 second of latency to every user though, and 99% probably aren't running into this perf stuff. what about 500ms? @tssweeney

Reply:
i think 500 is fine, sure. I general this live-filter effect is pretty rare in most applications. Instead there is a moment of "applying" the filter b/c this is so expensive.

Reply:
yeah true, i'll get the featurebee and you can play with 500